src/common/Gateway.sol

```diff
@@ -70,7 +70,12 @@ contract Gateway is Auth, Recoverable, IGateway {
     }
 
     modifier onlyManager(PoolId poolId) {
-        require(msg.sender == manager[poolId], ManagerNotAllowed());
+        require(manager[poolId][msg.sender], NotAuthorized());
+        _;
+    }
+
+    modifier onlyAuthOrManager(PoolId poolId) {
```

Comment:
This is actually a good point: we should make `blockOutgoing` into `onlyAuthOrManager` as well, for more flexibility.